app/policies/access_policy.rb

```diff
@@ -23,9 +23,26 @@ def configure
       end
     end
 
+    # This role can read published Kithe::Models.
     role :public do
-      can :read, Kithe::Model, { published: true }
+      can :read, Kithe::Model do |mod, user|
+        # mod could be any of Kithe::Model, Collection, Work, Asset,
+        # *or* an instance of any of the above classes.
+        #
+        # If mod is any of the above classes, definitely return false:
+        # the public is not allowed to read *all* possible instances of any of these classes.
+        #
+        # If mod is an instance, return true only if the instance is published.
+        mod.class < Kithe::Model && mod.published?
```

Comment:
Better way to write `mod.class < Kithe::Model` is just `model.instance_of?(Kithe::Model)`

Reply:
Done.

Reply:
Oops - not quite.
```
irb(main):004:0> goat = Asset.new()
=> #<Asset:0x00000001190f2070
...
irb(main):005:0> goat.instance_of? Asset
=> true
irb(main):006:0> goat.instance_of? Kithe::Model
=> false
irb(main):007:0> goat.class < Kithe::Model
=> true
```

Reply:
oops, you're right, `kind_of?`.  I could have sworn they were synonyms, but I guess I got confused! Good job double-chekcing me, thank you!